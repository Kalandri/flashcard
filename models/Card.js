const mongoose = require('mongoose');
const { Schema } = mongoose;

const cardSchema = new Schema({
	prompt: String,
	answer: String
});

module.exports = cardSchema;