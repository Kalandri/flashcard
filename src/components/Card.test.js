import React, { Componet } from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const props ={
	card: { prompt: 'test prompt', answer: 'test andwer'}
};

describe('Card', ()=>{
	const card = shallow(<Card {...props} />);

	it ('sets `reveal` to be `false`', ()=>{
		expect(card.state().reveal).toBe(false);
	});

	it('renders the card prompt', ()=>{
		expect(card.find('.card-prompt h4').text()).toEqual(props.card.prompt);
	});

	it('renders the card answer', ()=>{
		expect(card.find('.card-answer h4').text()).toEqual(props.card.answer);
	})
});
