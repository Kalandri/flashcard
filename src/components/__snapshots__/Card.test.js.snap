// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <div
    className="card-prompt"
  >
    <h4>
      test prompt
    </h4>
  </div>
  <div
    className="card-answer"
  >
    <h4
      className="text-hidden"
    >
      test answer
    </h4>
  </div>
</div>
`;
